src/core/Geometry.js

```diff
@@ -931,6 +944,17 @@ Object.assign( Geometry.prototype, EventDispatcher.prototype, {
 
 	},
 
+	setPrincipalUvsIndices: function( primaryUvsIndex, secondaryUvsIndex ) {
+
+		var uvsNeedUpdate = ( this.primaryUvsIndex !== primaryUvsIndex ) || ( this.secondaryUvsIndex !== secondaryUvsIndex);
+
+		this.primaryUvsIndex = primaryUvsIndex;
+		this.secondaryUvsIndex = secondaryUvsIndex;
+
+		this.uvsNeedUpdate = uvsNeedUpdate;
+		this.elementsNeedUpdate = uvsNeedUpdate;
```

Comment:
To be future proof, shouldnt this be

```javascript
if (uvsNeedUpdate) this.elementsNeedUpdate = true;
```

othervise it might cancel anything else that would cause it to need to be updated?